src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -207,8 +237,110 @@ public boolean equals(Object other) {
             return getName().equals(e.getName())
                     && getPhone().equals(e.getPhone())
                     && getSchool().equals(e.getSchool())
-                    && getYear().equals(e.getYear())
-                    && getAdmin().equals(e.getAdmin());
+                    && getYear().equals(e.getYear());
+        }
+    }
+
+    public static class EditAdminDescriptor {
+
+        private ClassTime time;
+        private ClassVenue venue;
+        private Fee fee;
+        private PaymentDate paymentDate;
+        private Set<AdditionalDetail> details;
+
+        public EditAdminDescriptor() {}
+
+        /**
+         * Copy constructor.
+         * A defensive copy of {@code details} is used internally.
+         */
+        public EditAdminDescriptor(EditAdminDescriptor toCopy) {
+            setTime(toCopy.time);
+            setVenue(toCopy.venue);
+            setFee(toCopy.fee);
+            setPaymentDate(toCopy.paymentDate);
+            setAdditionalDetails(toCopy.details);
+        }
+
+        /**
+         * Returns true if at least one field is edited.
+         */
+        public boolean isAnyFieldEdited() {
+            return CollectionUtil.isAnyNonNull(time, venue, fee, paymentDate, details);
+        }
+
+        public void setTime(ClassTime time) {
+            this.time = time;
+        }
+
+        public Optional<ClassTime> getClassTime() {
+            return Optional.ofNullable(time);
+        }
+
+        public void setVenue(ClassVenue venue) {
+            this.venue = venue;
+        }
+
+        public Optional<ClassVenue> getClassVenue() {
+            return Optional.ofNullable(venue);
+        }
+
+        public void setFee(Fee fee) {
+            this.fee = fee;
+        }
+
+        public Optional<Fee> getFee() {
+            return Optional.ofNullable(fee);
+        }
+
+        public void setPaymentDate(PaymentDate paymentDate) {
+            this.paymentDate = paymentDate;
+        }
+
+        public Optional<PaymentDate> getPaymentDate() {
+            return Optional.ofNullable(paymentDate);
+        }
+
+        /**
+         * Sets {@code details} to this object's {@code details}.
+         * A defensive copy of {@code details} is used internally.
+         */
+        public void setAdditionalDetails(Set<AdditionalDetail> details) {
+            this.details = (details != null) ? new HashSet<>(details) : null;
+        }
+
+        /**
+         * Returns an unmodifiable detail set, which throws {@code UnsupportedOperationException}
+         * if modification is attempted.
+         * Returns {@code Optional#empty()} if {@code details} is null.
+         */
+        public Optional<Set<AdditionalDetail>> getAdditionalDetails() {
+            return Optional.ofNullable(details).map(Collections::unmodifiableSet);
+            // return (details != null) ? Optional.of(Collections.unmodifiableSet(details)) : Optional.empty();
+        }
```

Comment:
Just to clarify for now we cant edit the additional details?

Reply:
Correct, we can't edit for now. I can do a separate PR to allow for such an edit

Reply:
okay thanks